utilities/cli/holohub.py

```diff
@@ -482,6 +571,17 @@ def _build_project_locally(
         if with_operators:
             cmake_args.append(f'-DHOLOHUB_BUILD_OPERATORS="{with_operators}"')
 
+        # Add mode-specific build dependencies
+        mode_config = self._get_mode_config(project_data, mode)
+        build_config = mode_config.get("build", {})
+
+        # Handle mode-specific build dependencies
+        if "depends" in build_config:
+            depends = build_config["depends"]
```

Comment:
Suggest `target_depends` or some other name to reflect more specific nature of this field reflecting op/app/ext dependencies only, open for discussion
```suggestion
        if "target_depends" in build_config:
            depends = build_config["target_depends"]
```